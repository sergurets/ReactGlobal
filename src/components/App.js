import React from "react";
import Item from "./item.js";
import Header from "./header.js";
import FullInfo from "./full_info.js";
import ErrorBoundary from "./error.js";
import { connect } from 'react-redux';
import { itemsFetchData } from '../actions/actions.js';
import { BrowserRouter as Router, Route, Link , Switch} from "react-router-dom";

import '../styles/App.css';


/*import * as React from 'react';
import * as ReactDom from 'react-dom';
import { BrowserRouter as Router, Route, NavLink, Switch } from 'react-router-dom';*/

const Posts = () => <p>Posts</p>;
const About = () => <p>About</p>;
const NotFound = () => <p>404 Not found</p>;

export class App extends React.Component {
  render() {
    return (
      <Router>
           <Switch>
            <Route exact path="/" component={Header} />
            <Route path="/posts" component={Posts} />
            <Route path="/about" component={About} />
            <Route path="/film/:id" component={FullInfo} />
            <Route path="/search" component={Header} />
            <Route path="*" component={NotFound} />
          </Switch>
      </Router>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
      fetchData: (url) => dispatch(itemsFetchData(url))
  };
};


export default connect(null, mapDispatchToProps)(App);
/*

ReactDom.render(<App />, document.getElementById('app'));


const Posts = () => <p>Posts</p>;
const About = () => <p>About</p>;
const NoMatch = () => <p>404</p>;

export class App extends React.Component {

  componentDidMount() {
    this.props.fetchData('http://react-cdp-api.herokuapp.com/movies?sortBy=release_date&sortOrder=desc')
  }

  render() {
    return (
      <div>
        <Link to="/posts">Posts</Link>
        <Link to="/films/447365">Film</Link>
         <Link to="/search/Search/54458477">Search</Link>
         <Switch>
           <Router>
          <Route path="/posts" component={Posts} />
          <Route path="/about" component={About} />
          <Route path="/films/:id" component={FullInfo} />
          <Route component={NoMatch} />
           </Router>
        </Switch>

      </div>
    );
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
      fetchData: (url) => dispatch(itemsFetchData(url))
  };
};


export default connect(null, mapDispatchToProps)(App);*/

// export default App;*/

/*
class App extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(value) {
    this.setState({ value: value })
  }

  componentDidMount() {
    this.props.fetchData('http://react-cdp-api.herokuapp.com/movies?sortBy=release_date&sortOrder=desc')
  }

  
  
  render() {
    return (
      <Router>
        <Link to="/about/">Home</Link>
        <Link to={`/film/${2}/`}>Films</Link>
          <Switch>
            <Route path="/film/" component={FullInfo} />  
            <Route path="/about/" component={Item} />  
          </Switch>
        {this.props.current ? (
        <div>
          <FullInfo/>
          <Item/>
        </div>) : (
        <div>
          <ErrorBoundary>
            <Header onClick = {this.onClick}/>
          </ErrorBoundary>
          <Item />
        </div>)}
      </Router>
    );
  }
}

const mapStateToProps = (state) => {
  return {
      current: state.current
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
      fetchData: (url) => dispatch(itemsFetchData(url))
  };
};


export default connect(mapStateToProps, mapDispatchToProps)(App);

// export default App;*/

