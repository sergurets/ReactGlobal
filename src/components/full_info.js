import React, { Component } from "react";

import '../styles/full_info.css';


function FullInfo(props) {
    var release_date = props.release_date ? props.release_date : '';
    if (release_date.length >4 ) {
      release_date = release_date.slice(0,4)
    }


  //  console.log(props)
    return (
        <div className="full_info">
          <div className="full_info_controls">
            <h2>Netflix search</h2>
            <button className="searchButton">Search</button>
          </div>
          <div className="full_info_wrapper">
            <img src={props.poster_path}/>
            <div className="full_info_content">
                <h1>{props.title}</h1>
                <div className="content_data">
                    <span className="">{release_date}</span>
                    {props.runtime ? <span className="">{props.runtime} min</span> : null}
                </div>
                <p className="">{props.overview}</p>

            </div>     
          </div>
        </div>  
    );
}

export default FullInfo;