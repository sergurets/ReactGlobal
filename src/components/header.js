import React from "react";
import Search from "./search.js";

import '../styles/header.css';

function Header() {
      return (
        <div className="header">
          <div className="header_content">
            <h2>Netflix search</h2>
            <h1>FIND YOUR MOVIE</h1>
            <Search/>
          </div>
        </div>
      );
  }
  
  export default Header;