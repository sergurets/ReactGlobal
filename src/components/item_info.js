import React from "react";


function ItemInfo(props) {
    const genres = props.genres.reduce((sum, current) => {return sum + current + " & "}, '').slice(0,-3);
    const release_date = parseInt(props.release_date);
    return (
        <div className="">
         <div className="info_title">
            <span>{props.title}</span>
            <span className="release_date">{release_date}</span>
         </div>
            <span className="info_genres">
            {genres}
            </span>
        </div>
    );
}

ItemInfo.defaultProps = {
    title: '',
    release_date: '0',
    genres: ['']

};

export {ItemInfo}