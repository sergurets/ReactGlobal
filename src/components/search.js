import React   from "react";
import { connect } from 'react-redux'
import { itemsFetchData, currentItem, sortItems } from '../actions/actions.js';
import  queryString  from 'query-string';

import '../styles/search.css';

class Search extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
         value: '',
         queryParam: Object.assign(
          {searchBy: 'title' ,
           sortBy: 'release_date',
           sortOrder: 'desc'},
           queryString.parse(props.location.search)
           )
        };
        
        this.handleChange = this.handleChange.bind(this);
        this.onClick = this.onClick.bind(this)
        this.keyPress = this.keyPress.bind(this);
        this.searchByTitle = this.searchByTitle.bind(this);
        this.searchByGenre = this.searchByGenre.bind(this);
        this.sortByReleaseDate=this.sortByReleaseDate.bind(this);
        this.sortByRating=this.sortByRating.bind(this);
      }

      onClick() {
        var url = `http://react-cdp-api.herokuapp.com/movies?${queryString.stringify(this.state.queryParam)}`;
        this.props.fetchData(url);
        this.props.current(null);
        this.props.history.push(`/search?${queryString.stringify(this.state.queryParam)}`)
      }

      componentDidMount(){
        const url = queryString.stringify(queryString.parse(this.props.location.search));
        url ? this.props.fetchData(`http://react-cdp-api.herokuapp.com/movies?${url}`) : null;
      }
    
      handleChange(event) {
        this.setState({
          value: event.target.value, 
          queryParam: Object.assign({},this.state.queryParam, {search: encodeURI(event.target.value)})
        }); 
      }

      searchByGenre() {
        this.setState({queryParam: Object.assign({},this.state.queryParam, {searchBy: 'genres'})})
      }

      searchByTitle() {
        this.setState({queryParam: Object.assign({},this.state.queryParam, {searchBy: 'title'})})
      }

      sortByReleaseDate() {
        this.setState({queryParam: Object.assign({},this.state.queryParam, {sortBy: 'release_date'})},
        () => this.props.sortBy(this.props.items, this.state.queryParam.sortBy))
      }

      sortByRating() {
        this.setState({queryParam: Object.assign({},this.state.queryParam, {sortBy: 'vote_average'})},
        () => this.props.sortBy(this.props.items, this.state.queryParam.sortBy))
      }

      keyPress(event){
        if(event.keyCode == 13){
          this.onClick();
        }
      } 

    render() {
      console.log('search', this, queryString.parse(this.props.location.search), this.props.location.search)
      var searchByTitle = this.state.queryParam.searchBy === 'title';
      var sortByDate = this.state.queryParam.sortBy === 'release_date';
      return (
        <div className="search">
          <div className="search_controls">
            <input
             id="mainSearch"
             type="text"
             value={this.state.value}
             onChange={this.handleChange}
             onKeyDown={this.keyPress}/>
            <div className="searchButtons">
              <div className="searchFilter">
                <span>SEARCH BY</span>
                <button 
                 onClick={this.searchByTitle}
                 className={searchByTitle ? "active" : ""}>
                 TITLE
                </button>
                <button 
                 onClick={this.searchByGenre}
                 className={searchByTitle ?  "" : "active"}>
                 GENRE
                </button>
              </div>
              <button className="mainSearchButton" onClick={this.onClick}>
               Search
              </button>
            </div>
          </div>

        <div className="sort_data">
          <div className="genres">
            <span>{this.props.total} movies found</span>
          </div>
          <div className="sort_buttons">
            <span>Sort by</span>
            <span 
              onClick={this.sortByReleaseDate}
              className={sortByDate ? "active" : ""}>
              release date
            </span>
            <span 
              onClick={this.sortByRating}
              className={sortByDate ? "" : "active"}>
              rating
            </span>
          </div>
          </div>  
        </div>
      );
    }
  }
  
const mapStateToProps = (state) => {
  return {
      total: state.items.total,
      items: state.items,
      hasErrored: state.itemsHasError,
      isLoading: state.itemsIsLoading
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
      fetchData: (url) => dispatch(itemsFetchData(url)),
      current: (item) => dispatch(currentItem(item)),
      sortBy: (item, sort) => dispatch(sortItems(item, sort))
  };
};


export default connect(mapStateToProps,  mapDispatchToProps)(Search);