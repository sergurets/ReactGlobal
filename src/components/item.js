import React from 'react';
import { initialSearchByGenres } from '../utilities/urls.ts';
import { ItemInfo } from './item_info';
import '../styles/item.css';

class Item extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(value) {
    this.props.current(value);
    this.props.fetchData(initialSearchByGenres(value));
  }

  render() {
    if (this.props.isLoading) return <h3>Loading...</h3>;
    if (this.props.hasErrored) return <h3>Error</h3>;
    if (!this.props.data.length) return <h3>No Films Found</h3>;
    let { data } = this.props;
    if (this.props.currentId) {
      data = data.filter(el => el.id !== this.props.currentId);
    }
    return (
          <div>
            {this.props.currentId ? null : <h1>Films</h1>}
            <div className="search_results">
                {data.map(el => (<div className="search_Item" key={el.id} onClick={this.onClick.bind(this, el)}>
                <img src={el.poster_path}/>
                <ItemInfo title={el.title} release_date={el.release_date} genres={el.genres}/>
                </div>
                ))}
            </div>
          </div>
    );
  }
}

Item.defaultProps = {
  data: [{
    title: '',
    id: '',
    poster_path: '',
    release_date: '0',
    genres: [''],
  }],
};

export { Item };
