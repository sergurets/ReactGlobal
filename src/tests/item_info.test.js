import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ItemInfo } from '../components/item_info';
describe('ItemInfo', () => {
  it('should render correctly', () => {
    const output = shallow(
      <ItemInfo/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
