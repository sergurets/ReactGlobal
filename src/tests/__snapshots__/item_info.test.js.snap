// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemInfo should render correctly 1`] = `
<div
  className=""
>
  <div
    className="info_title"
  >
    <span />
    <span
      className="release_date"
    >
          
    </span>
  </div>
  <span
    className="info_genres"
  />
</div>
`;
